# 2.4. Анализ рынка недвижимости
- Сколько в среднем стоят дома, у которых 2 спальни?
- Какая в среднем общая площадь домов(sqft_lot), у которых стоимость больше 600000?
- Как много домов коснулся ремонт? Как много домов не было отремонтировано? Используйте поле yr_renovated. Подсчитайте процентное отношение этих величин.
- Насколько в среднем стоимость домов с оценкой grade домов выше 10 отличается от стоимости домов с оценкой grade меньше 4?#£

In [2]:
import pandas as pd

In [3]:
df = pd.read_csv('kc_house_data.csv')

In [4]:
df.head(2)

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   

   sqft_lot15  
0        5650  
1        7639  

[2 rows x 21 columns]

Наименования столбцов и их суть:
- id - Уникальный ID для каждого дома
- date - Дата продажи дома
- price - Стоимость продажи дома
- bedrooms - Кол-во спален
- bathrooms - Кол-во ванных комнат (0.5 - туалет без душа)
- sqft_living - Кв. метры жилые
- sqft_lot - Кв. метры общие
- floors - Кол-во этажей
- waterfront - Есть набержная или нет
- view - Значение от 0 до 4 насколько хороший вид
- condition - Значение от 1 до 5 насколько хорошее состояние
- grade - Значение от 1 до 13, где 1-3 плохая конструкция здания и дизайн, 7 - средний уровень конструкции и дизайна, 11-13 - высокое качество конструкции и дизайна
- sqft_above - Кв. метры дома, которые находятся выше земли
- sqft_basement - Кв. метры дома, которые находятся ниже земли
- yr_built - Год постройки дома
- yr_renovated - Год ремонта дома
- zipcode - Индекс
- lat - Широтма
- long - Долгота
- sqft_living15 - Кв. метры жилой площади у 15 соседей
- sqft_lot15 - Кв. метры общей площади у 15 соседей

### Часть 1

In [12]:
df[df['bedrooms'] == 2]['price'].mean()

401372.681884058

Вывод: 401 372,68 - средняя стоимость дома с двумя спальнями

### Часть 2

In [17]:
df[df['price'] > 600000]['sqft_lot'].mean()

20442.524776214832

Вывод: средняя общая площадь домов, у которых стоимость больше 600000, составялет 20 442,52 квадратных метра

### Часть 3

In [19]:
df['yr_renovated'].unique()

array([   0, 1991, 2002, 2010, 1999, 1992, 2013, 1994, 1978, 2005, 2008,
       2003, 1984, 1954, 2014, 2011, 1974, 1983, 1945, 1990, 1988, 1957,
       1977, 1981, 1995, 2000, 1998, 1970, 1989, 2004, 1986, 2009, 2007,
       1987, 1973, 2006, 1985, 2001, 1980, 1971, 1979, 1997, 1950, 1969,
       1948, 2015, 1968, 2012, 1963, 1951, 1993, 1962, 1996, 1972, 1953,
       1955, 1982, 1956, 1940, 1976, 1946, 1975, 1958, 1964, 1959, 1960,
       1967, 1965, 1934, 1944])

In [33]:
non_renovated = df[df['yr_renovated'] == 0]['id'].count()
non_renovated

20699

In [34]:
renovated = df[~(df['yr_renovated'] == 0)]['id'].count()
renovated

914

In [35]:
all_houses = non_renovated + renovated
all_houses

21613

In [36]:
part_renovated = renovated * 100 / non_renovated
part_renovated

4.415672254698294

Вывод: ремонт коснулся 914 зданий, ремонт не был у 20 699 зданий: процент отремонтированных зданий равен всего 4,4.

### Часть 4

In [42]:
price_grade_10 = df[df['grade'] > 10]['price'].mean()
price_grade_10

1678635.1175298805

In [43]:
price_grade_4 = df[df['grade'] < 4]['price'].mean()
price_grade_4

189750.0

In [44]:
average_price = price_grade_10 - price_grade_4
average_price

1488885.1175298805

Вывод: в среднем на 1 488 885 составляет разница между средней стоимостью дома с оценкой grade домов выше 10 и средней стоимости дома с оценкой grade меньше 4